ha/setup/create_cluster.py

```diff
@@ -0,0 +1,92 @@
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import argparse
+from ha.execute import SimpleCommand
+from cortx.utils.log import Log
+
+def cluster_auth(username, password, nodelist):
+    """Authorizes cluster nodes.
+
+    Parameters:
+            username - just a user name to make cluster authorization
+            password - just a password for aforementioned user
+            nodelist - List with nodes which to authorize
+    Returns: None."""
+    nodes = " ".join(nodelist)
+    cmd_auth = f"pcs cluster auth -u {username} -p {password} {nodes}"
+    SimpleCommand().run_cmd(cmd_auth)
+
+def cluster_create(cluster_name, nodelist, start=False, enable=True):
+    """Creates cluster on given nodes. Enables and starts cluster if needed.
+
+    Parameters:
+        cluster_name    - name of the cluster to be created
+        nodelist        - List with nodes where setup the cluster
+        start           - whether cluster shall be started immediately
+        enable          - whether cluster service shall start on boot
+
+    Returns: None."""
+    nodes = " ".join(nodelist)
+    cmd_setup = f"pcs cluster setup --name {cluster_name} {nodes}"
+    cmd_stonith = "pcs property set stonith-enabled=False"
+    cmd_start = f"pcs cluster start {nodes}"
+    cmd_enable = f"pcs cluster enable {nodes}"
+
+    cmdlist = [ cmd_setup, cmd_stonith ]
+    if enable:
+        cmdlist.append(cmd_enable)
+    if start:
+        cmdlist.append(cmd_start)
+    for s in cmdlist:
+        SimpleCommand().run_cmd(s)
+
+def _parse_input_args():
+    """Parse and validate input arguments passed by mini-provisioner or CLI.
+
+    Returns dictonary with context."""
+    parser = argparse.ArgumentParser(description="Creates pacemaker cluster")
+    parser.add_argument("--cluster", default="cortx-lr2", type=str, help="Cluster name")
+    parser.add_argument("--username", required=True, type=str, help="User to be used for cluster auth")
+    parser.add_argument("--password", required=True, type=str, help="Password to be used for cluster auth")
+    parser.add_argument("--start", action='store_true', help="Start cluster on creation")
+    # Nodes to setup cluster can be entered via file or via parameters
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument("nodelist", type=str, nargs="*", default=[], help="List of nodes where cluster can be created")
+    group.add_argument("--nodefile", required=False, type=str, help="File with list of nodes")
+    return parser.parse_args()
+
+def _read_file_list(filename):
+    nodelist = []
+    with open(filename, "r") as f:
+        nodelist = f.read().splitlines()
+    return nodelist
+
+def _main():
+    # Workaround to make SimpleCommand work, not crash
+    Log.init(service_name="create_cluster",
+             log_path="/var/log/seagate/cortx/ha", level="INFO")
```

Comment:
log_path - Is this the path where ha_setup logs are being stored? If not, please talk to @indrajitzagade and set the path and logfile to same.

Reply:
Well, i would say this shall be read from HA configuration file. Current solution is not good. I will change it a bit later, since it doesn't look like high priority.
For M0 it corresponds to the same path which is going to be used in ha_setup.py
The thing is that this initialization makes sense only to standalone `create_cluster.py`. Indrajit will use Python API, so his configuration will be automatically fetched.

Reply:
This is going to be ha_setup logger.